core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PushAggregationIntoTableScan.java

```diff
@@ -225,11 +231,47 @@ private static AggregateFunction toAggregateFunction(Context context, Aggregatio
                 .map(symbol -> new Variable(symbol.getName(), context.getSymbolAllocator().getTypes().get(symbol)));
 
         return new AggregateFunction(
-                signature.getName(),
+                getResolvedName(signature.getName()),
                 signature.getReturnType(),
                 arguments.build(),
                 sortBy.orElse(ImmutableList.of()),
                 aggregation.isDistinct(),
                 filter);
     }
+
+    public static String getResolvedName(String name)
+    {
+        AggregationFunction variance =
```

Comment:
Can we have generic way of resolving the alias name ?

Reply:
@Praveen2112 Thanks for your comment ! I am not familiar with the part of SQL parsing. Currently I couldn't find out a way to get aggregation function alias name without specifically point out aggregation method/class which contain alias. 
I've thought about parsing all classes and methods under package `io.trino.operator.aggregation` and get those who contains alias name, but I'm not sure if this is a good idea, or maybe there is another better way.

Gentle ping @wendigo.

Reply:
Finally figuring out another way to resolve alias name, could you take a look ? Many thanks ! 😃